### PR TITLE
GDB-10263: fixes help functionality

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
- When the user is on the "user" or "server" files import page and opens the help dialog, it disappears after a while;
- When the import page is opened, two jobs that update the imported resource list are started.

## Why
- There are changes in the requirement for when the help message should be displayed, during implementation this functionality was broken;
- The import view has a main controller, one controller for the import tab, and one for the server tab. Both tab controllers extend the main controller. When the page is opened, both tab controllers are initialized, and each one starts a job to update the list of resources.

## How
- Fixed the condition for when the help message should be displayed.
- Added "ng-if" conditions in the template to control when the controller for the user tab or server tab is initialized. Now, only one is initialized, and only one job runs.

### CODE REVIEW

File: src/js/angular/import/controllers/import-view.controller.js [hunk 1]
```diff
@@ -86,6 +87,7 @@ importViewModule.controller('ImportViewCtrl', ['$scope', 'toastr', '$interval',
         $scope.toTitleCase = (str) => stringUtils.toTitleCase(str);
 
         $scope.pollList = () => {
+            console.log("polling is created")
```
Comment (svilenvelikov) [inline src/js/angular/import/controllers/import-view.controller.js:90]: loggers should be removed

File: src/js/angular/import/controllers/tab.controller.js [hunk 2]
```diff
@@ -25,38 +25,39 @@ function TabController($scope, $location, ImportViewStorageService, ImportContex
...
-        setIsHelpVisible(isVisible);
-    };
-
-    const onResourcesUpdated = (resources) => {
-        setHelpVisibility(resources);
-    };
+    const init = () => {
+        Object.values(TABS).forEach((tabId) => checkEmptyState[tabId] = true);
+        // // When controller is initialized we reset the state of persistence. Initially the help have to be opened.
+        // ImportViewStorageService.setHelpVisibility($scope.activeTabId.false);
```
Comment (svilenvelikov) [inline src/js/angular/import/controllers/tab.controller.js:55]: delete commented code

File: src/js/angular/models/import/import-view-persistence.js [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
-        this._isHelpVisible = !this._isHelpVisible;
+    toggleHelpVisibility(tabId) {
+        this.setIsHelpVisible(tabId, !this._persistence[tabId].isHelpVisible);
     }
 
     toJSON() {
-        return {
-            isHelpVisible: this._isHelpVisible
-        };
+        return JSON.stringify(this._persistence);
```
Comment (svilenvelikov) [inline src/js/angular/models/import/import-view-persistence.js:25]: This is incorrect. The method is called `toJSON` and not `toString`

File: src/js/angular/import/controllers/tab.controller.js [hunk 4]
```diff
@@ -25,38 +25,39 @@ function TabController($scope, $location, ImportViewStorageService, ImportContex
     // Public functions
     // =========================
 
+    const updateHelpVisibility = () => {
+        let empty = checkEmptyState[$scope.activeTabId] && areResourcesEmpty();
+        let helpVisible = ImportViewStorageService.getIsHelpVisible($scope.activeTabId);
```
Comment (svilenvelikov) [inline src/js/angular/import/controllers/tab.controller.js:30]: you should have lint warnings here ^. empty and helpVisible should be `const` and not `let`

File: test-cypress/integration/import/import-view.spec.js [hunk 5]
```diff
@@ -42,4 +44,58 @@ describe('Import view', () => {
         ImportUserDataSteps.getResources().should('have.length', 1);
         ImportUserDataSteps.getResourceByName('bnodes.ttl').should('be.visible');
     });
+
+    it.only('Should display/hide help message depends on resource result', () => {
```
Comment (svilenvelikov) [inline test-cypress/integration/import/import-view.spec.js:48]: only

File: test-cypress/steps/import/import-steps.js [hunk 6]
```diff
@@ -143,7 +143,7 @@ class ImportSteps {
         return cy.window().its('navigator.clipboard').invoke('readText').then((text) => text);
     }
 
-    static openHelpMessage() {
+    static toggleHelpMessage() {
         // For some reason the page info box opens unexpectedly and covers the help info icon.
         this.getView().find('.toggle-help-btn').click({force: true});
```
Comment (svilenvelikov) [inline test-cypress/steps/import/import-steps.js:148]: is the force still needed?

Comment (boyan-tonchev) [reply test-cypress/steps/import/import-steps.js:148]: I will check

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**